app/src/docs/v1.api-spec.yaml

```diff
@@ -385,65 +585,151 @@ components:
         subject:
           type: string
           description: The email subject
-          example: 'Hello {{ someone }}'
-    EndpointStatus:
+          example: Hello user
+        to:
+          type: array
+          items:
+            type: string
+          description: >-
+            An array of recipients email addresses that will appear on the To:
+            field
+          example:
+            - baz@gov.bc.ca
+    NotFound:
+      allOf:
+        - $ref: '#/components/schemas/Problem'
+        - type: object
+          properties:
+            status:
+              example: 404
+            title:
+              example: Not Found
+            type:
+              example: 'https://httpstatuses.com/404'
+    Problem:
       type: object
       required:
-        - name
-        - endpoint
-        - healthCheck
-        - authenticated
-        - authorized
+        - status
+        - title
+        - type
       properties:
-        name:
+        details:
           type: string
-          description: The name of the endpoint
-          example: WebADE API
-        endpoint:
+          description: contains a message about the failure
+          example: Error message
+        status:
+          type: integer
+          description: http status code
+        title:
           type: string
-          description: Base endpoint URL
-          example: 'https://example.com/v1/'
-        healthCheck:
-          type: boolean
-          description: Is the endpoint reachable
-        authenticated:
-          type: boolean
-          description: Are credentials valid to access endpoint
-        authorized:
-          type: boolean
-          description: Do the credentials have the right permissions
-    EndpointList:
+          description: http status description
+        type:
+          type: string
+          description: reference for the error
+    StatusObject:
       type: object
-      required:
-        - endpoints
       properties:
-        endpoints:
+        delayUntil:
+          type: integer
+          description: UTC time to delay queueing the message for sending
+          example: 1570000000
+        log:
```

Comment:
Don't like this.  Make it a status history and it's an array of status codes and timestamps - what state it was at a given time.  Log is weird, too low-level and implementation based concept.

Reply:
Added tag to ContextObject and MessageObject schemas. We may need to further discuss the log array. I'll investigate what we have available from Bull and try to think of some options.

Reply:
Bull should have no bearing on our API.  We translate Bull into our business; Bull is replaceable.  Callers should easily be able to discern what states this message has been in.  So list of our business statuses and the time they entered that state.

Reply:
As discussed will leave out log array until further implementation.

Reply:
Log has been dropped in latest commit.